lib/compute-at-edge-abi/typenames.witx

```diff
@@ -94,6 +94,8 @@
 (typename $dictionary_handle (handle))
 ;;; A handle to an Object Store.
 (typename $object_store_handle (handle))
+;;; A handle to a pending Object Store request.
+(typename $pending_object_store_handle (handle))
```

Comment:
We might want to call this `$pending_object_store_lookup_handle`. I realize that's verbose, but it makes room for other async operations on the object story. (And, with the impending rename, it'd be `$pending_kv_lookup_handle` which isn't too bad.)

Reply:
good thinking, the store has other methods such as list keys and metadata which would be good to have async when they are implemented 👍

Reply:
Done :+1: